### PR TITLE
Add tables to utility page

### PR DESCRIPTION
**Pull request by krofax**
Added tables to utility and paymasters utility page.

### CODE REVIEW

File: docs/api/js/paymaster-utils.md [hunk 1]
```diff
@@ -18,6 +18,12 @@ export const IPaymasterFlow = new ethers.utils.Interface(require('../../abi/IPay
 
 Returns encoded input for an approval-based paymaster.
 
+#### Inputs
+
+| Parameter           | Type                    | Description                     |
+| ------------------- | ----------------------- | ------------------------------- |
+| `paymasterInput`    | `BytesLike`             | The input data to the paymaster.|
```
Comment (kailash-manasarovar) [inline docs/api/js/paymaster-utils.md:25]: this is not a BytesLike

File: docs/api/js/paymaster-utils.md [hunk 2]
```diff
@@ -31,6 +37,11 @@ export function getApprovalBasedPaymasterInput(paymasterInput: ApprovalBasedPaym
 ### `getGeneralPaymasterInput`
 
 As above but for general-based paymaster.
+#### Inputs
+
+| Parameter           | Type                    | Description                     |
+| ------------------- | ----------------------- | ------------------------------- |
+| `paymasterInput`    | `BytesLike`             | The input data to the paymaster.|
```
Comment (kailash-manasarovar) [inline docs/api/js/paymaster-utils.md:44]: nor this

File: docs/api/js/paymaster-utils.md [hunk 3]
```diff
@@ -42,8 +53,17 @@ export function getGeneralPaymasterInput(paymasterInput: GeneralPaymasterInput):
 
 Returns a correctly-formed `paymasterParams` object for common [paymaster flows](../../dev/developer-guides/aa.md#built-in-paymaster-flows).
 
+
+#### Inputs
+
+| Parameter           | Type                    | Description                       |
+| ------------------- | ----------------------- | --------------------------------  |
+| `paymasterAddress`  | `Address` string        | The non-zero `paymaster` address. |
+| `paymasterInput`    | `BytesLike`             | The input data to the paymaster.  |
```
Comment (kailash-manasarovar) [inline docs/api/js/paymaster-utils.md:62]: nor this

File: docs/api/js/utils.md [hunk 4]
```diff
@@ -168,7 +168,9 @@ Converts the address that submitted a transaction to the inbox on L1 to the `msg
 
 #### Inputs
 
-- `address`: sender address as string.
+| Parameter | Type          | Description     |
+| --------- | ------------- | --------------- |
+| `address` | string        | Token address.  |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:173]: this is not a token address

File: docs/api/js/utils.md [hunk 5]
```diff
@@ -228,9 +232,10 @@ Generates a contract address from deployer's account and nonce.
 
 #### Inputs
 
-- `sender`: sender address as string.
-- `senderNonce`: sender nonce as `BigNumberish` object.
-
+| Parameter      | Type                  | Description      |
+| -------------- | --------------------- | ---------------- |
+| `sender`       | string                | Senders address. |
+| `senderNonce`  | `BigNumberish` object | Senders nonce.   |
```
[the remark this answers is not in the document]
Comment (kailash-manasarovar) [reply docs/api/js/utils.md:238]: if you're going to use the possessive, do it correctly

File: docs/api/js/utils.md [hunk 6]
```diff
@@ -280,15 +287,17 @@ Used by `estimateDefaultBridgeDepositL2Gas` to estimate L2 gas required for toke
...
+| `providerL2`        | object                           | zkSync Provider.                                             |
+| `l1BridgeAddress`   | string                           | Bridge Address.                                              |
+| `l2BridgeAddress`   | string                           | Bridge Address.                                              |
+| `token`             | string                           | Token Address.                                               |
+| `amount`            | `BigNumberish` object            | Deposit amount.                                              |
+| `to`                | string                           | Recipient Address.                                           |
+| `bridgeData`        | `BytesLike` object               | Data.                                                        |
+| `from?`             | string (optional)                | Sender Address.                                              |
+| `gasPerPubdataByte?`| `BigNumberish` object (optional) | Current gas per byte of pubdata.                             |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:300]:
```suggestion
| Parameter           | Type                             | Description                                                  |
| ------------------- | -------------------------------  | ------------------------------------------------------------ |
| `providerL2`        | object                           | zkSync provider.                                             |
| `l1BridgeAddress`   | string                           | L1 bridge address.                                              |
| `l2BridgeAddress`   | string                           | L2 bridge address.                                              |
| `token`             | string                           | Token address.                                               |
| `amount`            | `BigNumberish` object            | Deposit amount.                                              |
| `to`                | string                           | Recipient address.                                           |
| `bridgeData`        | `BytesLike` object               | Data.                                                        |
| `from?`             | string (optional)                | Sender address.                                              |
| `gasPerPubdataByte?`| `BigNumberish` object (optional) | Current gas per byte of pubdata.                             |
```

File: docs/api/js/utils.md [hunk 7]
```diff
@@ -322,13 +331,15 @@ Returns an estimation of L2 gas required for token bridging via the default ERC2
...
-- `gasPerPubdataByte?`: current gas per byte of pubdata as `BigNumberish` (optional).
+| Parameter           | Type                             | Description                                   |
+| ------------------- | -------------------------------- | --------------------------------------------- |
+| `providerL1`        | object                           | ethers Provider.                              |
+| `providerL2`        | string                           | zkSync Provider.                              |
+| `token`             | string                           | Token Address.                                |
+| `amount`            | `BigNumberish` object            | Deposit amount.                               |
+| `to`                | string                           | Recipient Address.                            |
+| `from?`             | string (optional)                | Sender Address.                               |
+| `gasPerPubdataByte?`| `BigNumberish` object  (optional)| Current gas per byte of pubdata.              |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:342]:
```suggestion
| Parameter           | Type                             | Description                                   |
| ------------------- | -------------------------------- | --------------------------------------------- |
| `providerL1`        | Provider object                           | Ethers provider.                              |
| `providerL2`        | Provider object                         | zkSync provider.                              |
| `token`             | string                           | Token address.                                |
| `amount`            | `BigNumberish` object            | Deposit amount.                               |
| `to`                | string                           | Recipient address.                            |
| `from?`             | string (optional)                | Sender address.                               |
| `gasPerPubdataByte?`| `BigNumberish` object  (optional)| Current gas per byte of pubdata.              |
```

File: docs/api/js/utils.md [hunk 8]
```diff
@@ -378,7 +389,9 @@ Returns a log containing details of all deployed contracts related to a transact
 
 #### Inputs
 
-- `receipt`: transaction receipt as ethers Provider object.
+| Parameter           | Type                         | Description                                                  |
+| ------------------- | ---------------------------  | ------------------------------------------------------------ |
+| `receipt`           | object                       | Transaction receipt.                                         |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:394]:
```suggestion
| `receipt`           | TransactionReceipt object                     | Transaction receipt.                                         |
```

File: docs/api/js/utils.md [hunk 9]
```diff
@@ -411,11 +424,13 @@ Returns the calldata sent by an L1 ERC20 bridge to its L2 counterpart during tok
...
-- `l2Receiver`: recipient address on L2 as string.
-- `amount`: gas fee for the number of tokens to bridge as `BigNumberish` object.
-- `bridgeData`: data as `BytesLike` object.
+| Parameter           | Type                         | Description                                   |
+| ------------------- | ---------------------------  | --------------------------------------------- |
+| `l1TokenAddress`    | string                       | Token address on L1.                          |
+| `l1Sender`          | string                       | Sender address on L1.                         |
+| `l2Receiver`        | string                       | Recipient address on L2.                      |
+| `amount`            | `BigNumberish` object        | Gas fee for the number of tokens to bridge.   |
+| `bridgeData`        | `BytesLike` object           | Data .                                        |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:433]:
```suggestion
| `bridgeData`        | `BytesLike` object           | Data                                        |
```

File: docs/api/js/utils.md [hunk 10]
```diff
@@ -441,9 +456,10 @@ Returns the data needed for correct initialization of an L1 token counterpart on
 
 #### Inputs
 
-- `l1TokenAddress`: token address on L1 as string.
-- `provider`: ethers `Provider` object.
-
+| Parameter           | Type                             | Description                                   |
+| ------------------- | -------------------------------- | --------------------------------------------- |
+| `l1TokenAddress`    | string                           | Token address on L1.                          |
+| `provider`          | object                           | Ethers Provider.                              |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:462]:
```suggestion
| `provider`          | Provider object                           | Ethers provider.                              |
```

File: docs/api/js/utils.md [hunk 11]
```diff
@@ -479,8 +495,10 @@ Returns the hash of the L2 priority operation from a given transaction receipt a
 
 #### Inputs
 
-- `txReceipt`: receipt of the L1 transaction as ethers `TransactionReceipt` object.
-- `zkSyncAddress`: address of zkSync Era main contract as `Address` string.
+| Parameter           | Type                        | Description                          |
+| ------------------- | --------------------------- | ------------------------------------ |
+| `txReceipt`         | `TransactionReceipt` object | Receipt of the L1 transaction.       |
+| `zkSyncAddress`     | `Address` string            | Address of zkSync Era main contract. |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:501]:
```suggestion
| `zkSyncAddress`     | `Address` as string            | Address of zkSync Era main contract. |
```

File: docs/api/js/utils.md [hunk 12]
```diff
@@ -514,9 +532,11 @@ Returns a keccak encoded message with a given sender address and block number fr
 
 #### Inputs
 
-- `sender`: the sender of the message on L2 as `Address` string.
-- `msg`: encoded message as `BytesLike` object.
-- `txNumberInBlock`: index of the transaction in the block as number.
+| Parameter           | Type                | Description                           |
+| ------------------- | ------------------- | ------------------------------------- |
+| `sender`            | `Address` string    | The sender of the message on L2.      |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:537]:
```suggestion
| `sender`            | `Address` as string    | The sender of the message on L2.      |
```

File: docs/api/js/utils.md [hunk 13]
```diff
@@ -608,10 +632,13 @@ Called from [`isSignatureCorrect`](#isSignatureCorrect) for contract account add
...
 #### Inputs
 
-- `provider`: `Provider` object.
-- `address`: sender address as string.
-- `msgHash`: hash of the message as string. 
-- `signature`: ethers signature as `SignatureLike` object. 
+
+| Parameter           | Type                    | Description                  |
+| ------------------- | ----------------------- | ---------------------------  |
+| `provider`          | `Provider` object       | The RPC provider.            |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:638]:
```suggestion
| `provider`          | `Provider` object       | Provider.            |
```

File: docs/api/js/utils.md [hunk 14]
```diff
@@ -608,10 +632,13 @@ Called from [`isSignatureCorrect`](#isSignatureCorrect) for contract account add
...
+| `address`           | string                  | Sender address as string.    |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:639]:
```suggestion
| `address`           | string                  | Sender address.    |
```

File: docs/api/js/utils.md [hunk 15]
```diff
@@ -637,7 +664,9 @@ Returns true if token represents ETH on L1 or L2.
 
 #### Inputs
 
-- `token`: `Address` object.
+| Parameter           | Type                    | Description                  |
+| ------------------- | ----------------------- | ---------------------------  |
+| `token`             | `Address` object        | The token adress.            |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:669]:
```suggestion
| `token`             | `Address` object        | The token address.            |
```

File: docs/api/js/utils.md [hunk 16]
```diff
@@ -651,10 +680,12 @@ Returns true if account abstraction EIP712 signature is correct.
 
 #### Inputs
 
-- `provider`: `Provider` object.
-- `address`: sender address as string.
-- `message`: string or Bytes object representation of the message. 
-- `signature`: ethers signature as `SignatureLike` object.
+| Parameter           | Type                    | Description                  |
+| ------------------- | ----------------------- | ---------------------------  |
+| `provider`          | `Provider` object       | The RPC provider.            |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:685]:
```suggestion
| `provider`          | `Provider` object       | Provider.            |
```

File: docs/api/js/paymaster-utils.md (hunk 1 above)
Comment (krofax) [reply docs/api/js/paymaster-utils.md:25]: @kailash-manasarovar, what could it be? Cause the [paymasterparams](https://era.zksync.io/docs/api/js/types.html#paymasterparams) page says it is.

Comment (krofax) [reply docs/api/js/paymaster-utils.md:25]: @kailash-manasarovar, what could it be? Cause the [paymasterparams](https://era.zksync.io/docs/api/js/types.html#paymasterparams) page says it is.

File: docs/api/js/utils.md [hunk 17]
```diff
@@ -674,10 +705,12 @@ Called from [`isMessageSignatureCorrect`](#ismessagesignaturecorrect) and [`isTy
...
-- `provider`: `Provider` object.
-- `address`: sender address as string.
-- `msgHash`: hash of the message as string. 
-- `signature`: ethers signature as `SignatureLike` object.
+| Parameter           | Type                    | Description                  |
+| ------------------- | ----------------------- | ---------------------------  |
+| `provider`          | `Provider` object       | The RPC provider.            |
+| `address`           | string                  | Sender address as string.    |
+| `msgHash`           | string                  | The hash of the message.     |
+| `signature`         | `SignatureLike` object  | Ethers signature.            |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:713]:
```suggestion
| `provider`          | `Provider` object       | Provider.            |
| `address`           | string                  | Sender address.    |
| `msgHash`           | string                  | The hash of the message.     |
| `signature`         | `SignatureLike` object  | Ethers signature.            |
```

File: docs/api/js/utils.md [hunk 18]
```diff
@@ -705,12 +738,14 @@ Returns true if account abstraction EIP712 signature is correct.
...
-- `value`: a single `Record` value as `Record<string, any>`.
-- `signature`: ethers signature as `SignatureLike` object.
+| Parameter           | Type                       | Description                                                 |
+| ------------------- | -------------------------- | ----------------------------------------------------------  |
+| `provider`          | `Provider` object          | The RPC provider.                                           |
+| `address`           | string                     | Sender address as string.                                   |
+| `domain`            | `TypedDataDomain` object   |                                                             |
+| `types`             | string                     | The map of records pointing from field name to field type.  |
+| `value`             | string                     | A single record.                                            |
+| `signature`         | `SignatureLike` object     | Ethers signature.                                           |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:748]:
```suggestion
| `provider`          | `Provider` object          | Provider.                                           |
| `address`           | string                     | Sender address.                                   |
| `domain`            | `TypedDataDomain` object   |       Domain data.                                                      |
| `types`             | Map<string, TypedDataField>    | Map of records pointing from field name to field type.  |
| `value`             | Record<string, any>                     | A single record value.                                           |
| `signature`         | `SignatureLike` object     | Ethers signature.                                           |
```

File: docs/api/js/utils.md [hunk 19]
```diff
@@ -744,7 +779,9 @@ Common sleep function that pauses execution for a number of milliseconds.
 
 #### Inputs
 
-- `millis`: number of milliseconds as number.
+| Parameter           | Type        | Description                                                        |
+| ------------------- | ----------- | -----------------------------------------------------------------  |
+| `millis`            | number      | The number of time required in milliseconds to pause an execution. |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:784]:
```suggestion
| `millis`            | number      | Number of milliseconds. |
```

File: docs/api/js/paymaster-utils.md (hunk 1 above)
Comment (kailash-manasarovar) [reply docs/api/js/paymaster-utils.md:25]: check the code

Comment (krofax) [reply docs/api/js/paymaster-utils.md:25]: Okay, but if you check the [overrides](https://era.zksync.io/docs/api/js/features.html#overrides), you see this:

```
{
    overrides: {
        customData: {
            gasPerPubdata?: BigNumberish;
            factoryDeps?: BytesLike[];
            customSignature?: BytesLike;
            paymasterParams?: {
                paymaster: Address;
                paymasterInput: BytesLike;
            };
        }
    }
}

```
**paymasterInput** type is defined as `BytesLike`.

I'll confirm the type with a dev.

File: docs/api/js/utils.md [hunk 20]
```diff
@@ -280,15 +287,17 @@ Used by `estimateDefaultBridgeDepositL2Gas` to estimate L2 gas required for toke
...
-- `l2BridgeAddress`: bridge `Address` string.
-- `token`: token `Address` string.
-- `amount`: deposit amount as `BigNumberish`.
-- `to`: recipient `Address` string.
-- `bridgeData`: data as `BytesLike` object.
-- `from?`: sender `Address` string (optional).
-- `gasPerPubdataByte?`: current gas per byte of pubdata as `BigNumberish` (optional).
+| Parameter           | Type                             | Description                                                  |
+| ------------------- | -------------------------------  | ------------------------------------------------------------ |
+| `providerL2`        | object                           | zkSync provider.                                             |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:292]:
```suggestion
| `providerL2`        | `Provider` object                           | zkSync provider.                                             |
```

File: docs/api/js/utils.md [hunk 21]
```diff
@@ -378,7 +389,9 @@ Returns a log containing details of all deployed contracts related to a transact
 
 #### Inputs
 
-- `receipt`: transaction receipt as ethers Provider object.
+| Parameter           | Type                         | Description                                                  |
+| ------------------- | ---------------------------  | ------------------------------------------------------------ |
+| `receipt`           | TransactionReceipt object                     | Transaction receipt.                                         |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:394]:
```suggestion
| `receipt`           | `TransactionReceipt` object                     | Transaction receipt.                                         |
```

File: docs/api/js/utils.md [hunk 22]
```diff
@@ -280,15 +287,17 @@ Used by `estimateDefaultBridgeDepositL2Gas` to estimate L2 gas required for toke
...
-- `gasPerPubdataByte?`: current gas per byte of pubdata as `BigNumberish` (optional).
+| Parameter           | Type                             | Description                                                  |
+| ------------------- | -------------------------------  | ------------------------------------------------------------ |
+| `providerL2`        | object                           | zkSync provider.                                             |
+| `l1BridgeAddress`   | string                           | L1 bridge address.                                              |
+| `l2BridgeAddress`   | string                           | L2 bridge address.                                              |
+| `token`             | string                           | Token address.                                               |
+| `amount`            | `BigNumberish` object            | Deposit amount.                                              |
+| `to`                | string                           | Recipient address.                                           |
+| `bridgeData`        | `BytesLike` object               | Data.                                                        |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:298]:
```suggestion
| `bridgeData`        | `BytesLike` object               | Bridge data.                                                        |
```

File: docs/api/js/utils.md [hunk 23]
```diff
@@ -705,12 +738,14 @@ Returns true if account abstraction EIP712 signature is correct.
...
-- `domain`: `TypedDataDomain` object from `@ethersproject/abstract-signer`.
-- `types`: map of records pointing from field name to field type as `Record<string, Array<TypedDataField>>`. 
-- `value`: a single `Record` value as `Record<string, any>`.
-- `signature`: ethers signature as `SignatureLike` object.
+| Parameter           | Type                       | Description                                                 |
+| ------------------- | -------------------------- | ----------------------------------------------------------  |
+| `provider`          | `Provider` object          | Provider.                                           |
+| `address`           | string                     | Sender address.                                   |
+| `domain`            | `TypedDataDomain` object   |       Domain data.                                                      |
+| `types`             | Map<string, TypedDataField>    | Map of records pointing from field name to field type.  |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:746]:
```suggestion
| `types`             | `Map<string, TypedDataField>`    | Map of records pointing from field name to field type.  |
```

File: docs/api/js/utils.md [hunk 24]
```diff
@@ -705,12 +738,14 @@ Returns true if account abstraction EIP712 signature is correct.
...
+| `value`             | Record<string, any>                     | A single record value.                                           |
```
Comment (kailash-manasarovar) [inline docs/api/js/utils.md:747]:
```suggestion
| `value`             | `Record<string, any>`                     | A single record value.                                           |
```

**Comment by krofax** [comment]
@kailash-manasarovar good to go?

**Review by kailash-manasarovar — APPROVED** [review]
LGTM

**Pull request merged**